Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -28,39 +29,61 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
             for pair in declaration {
                 let (platforms, declaration) = pair
                 
-                let renderedTokens = declaration.declarationFragments.map { token -> DeclarationRenderSection.Token in
-                    
-                    // Create a reference if one found
-                    var reference: ResolvedTopicReference?
-                    if let preciseIdentifier = token.preciseIdentifier,
-                       let resolved = renderNodeTranslator.context.symbolIndex[preciseIdentifier] {
-                        reference = resolved
-                        
-                        // Add relationship to render references
-                        renderNodeTranslator.collectedTopicReferences.append(resolved)
-                    }
-                    
-                    // Add the declaration token
-                    return DeclarationRenderSection.Token(fragment: token, identifier: reference?.absoluteString)
-                }
-                
                 let platformNames = platforms.sorted { (lhs, rhs) -> Bool in
                     guard let lhsValue = lhs, let rhsValue = rhs else {
                         return lhs == nil
                     }
                     return lhsValue.rawValue < rhsValue.rawValue
                 }
                 
+                let renderedTokens = renderDeclarationTokens(fragments: declaration.declarationFragments, 
+                                                             renderNodeTranslator: &renderNodeTranslator)
+
+                // If this symbol has overloads, render their declarations as well.
+                let overloadsVariant = symbol.overloadsVariants[trait]
```

Comment:
Maybe extra the following lines of code into a separate method similar to `renderDeclarationTokens` ? Then you could simplify this method down substantially into a single call to `declarations.append(....`